addons/purchase/models/purchase.py

```diff
@@ -484,6 +490,33 @@ def _compute_qty_invoiced(self):
                         qty -= inv_line.uom_id._compute_quantity(inv_line.quantity, line.product_uom)
             line.qty_invoiced = qty
 
+    @api.multi
+    @api.depends('product_id')
```

Comment:
Isn't it product_id.type ?

Reply:
I think, if you are changing the type of a product (which is a bad idea, and prevent in most of cases I think), you don't want to retrigger the computation of receive_qty field of all the purchase order lines related to this product.
And we already applied the same strategy for sale (delivered_qty) in stock and timesheet : https://github.com/odoo/odoo/blob/12.0/addons/sale_stock/models/sale_order.py#L168